repentogon/resources/scripts/main_ex.lua

```diff
@@ -1407,6 +1412,57 @@ function _RunPostPickupSelection(callbackID, param, pickup, variant, subType, ..
 	return recentRet
 end
 
+local preStatusApplyTypes = {
+	[StatusEffect.CONFUSION] = checkTableTypeFunction({ "integer", "boolean" }),
+	[StatusEffect.KNOCKBACK] = checkTableTypeFunction({ "integer", "Vector", "boolean" }),
+	[StatusEffect.SLOWING] =checkTableTypeFunction({ "integer", "number", "Color" }),
+	[StatusEffect.POISON] = checkTableTypeFunction({ "integer", "number" }),
+	[StatusEffect.BURN] = checkTableTypeFunction({ "integer", "number" }),
+}
+
+-- Custom handling for MC_PRE_STATUS_EFFECT_APPLY
+-- Handle type checking here since the callback is called for several status effects with different types,
+-- terminate early if false is returned,
+-- and use additive callback logic instead of the default one.
+local function RunPreStatusEffectApplyCallback(callbackID, param, status, entity, entityRef, duration, extraParam1, extraParam2)
+	local recentRet = nil
+
+	for callback in GetCallbackIterator(callbackID, param) do
+		local ret = RunCallbackInternal(callbackID, callback, status, entity, entityRef, duration, extraParam1, extraParam2)
+
+		if type(ret) == "boolean" then
+			if ret == false then return false end
+			if recentRet == nil or type(recentRet) == "boolean" then 
+				recentRet = ret
+			end
+		elseif type(ret) == "table" then
+			if preStatusApplyTypes[status] then
+				local err = preStatusApplyTypes[status](ret)
+
+				if err then
+					logError(callbackID, callback.Mod.Name, err)
+				else
+					recentRet = ret
+					duration = ret[1]
```

Comment:
I believe that returning an empty table or a table with only one element (the duration) will pass the type check, as the typecheck function only verifies that any values presents are of expected types, and doesn't care about missing elements.

If you want to allow incomplete table returns, you'll need to merge returned tables with previously returned tables, and account for null entries when updating the extraParams.

Alternatively, you could require that the returned table is always of the exact expected size.

Reply:
Made the returned table need to have a minimum size